pulp_ansible/app/galaxy/v3/views.py

```diff
@@ -295,14 +299,18 @@ def update(self, request, *args, **kwargs):
         serializer = self.get_serializer(collection, data=request.data, partial=True)
         serializer.is_valid(raise_exception=True)
         deprecated_value = request.data.get("deprecated")
-        if deprecated_value:
-            AnsibleCollectionDeprecated.objects.get_or_create(
-                repository_version=repo_version, collection=collection
-            )
-        else:
-            AnsibleCollectionDeprecated.objects.filter(
-                repository_version=repo_version, collection=collection
-            ).delete()
+        AnsibleCollectionDeprecated.objects.get_or_create(
+            namespace=collection.namespace, name=collection.name
+        )
+        deprecation = AnsibleCollectionDeprecated.objects.filter(
+            namespace=collection.namespace, name=collection.name
+        )
+        with repo_version.repository.new_version() as new_version:
```

Comment:
This has to use a task because it's making a new repository version. I think it should use a brand new task.